tests/contrib/httpx/test_httpx.py

```diff
@@ -140,84 +138,77 @@ def assert_spans(test_spans, service):
     assert_spans(test_spans, service="async-client")
 
 
-def test_configure_service_name_env(run_python_code_in_subprocess):
+@pytest.mark.subprocess(
+    env=dict(
+        DD_HTTPX_SERVICE="env-overridden-service-name",
+        DD_SERVICE="global-service-name",
+    )
+)
+def test_configure_service_name_env():
     """
     When setting DD_HTTPX_SERVICE env variable
         When DD_SERVICE is also set
             We use the value from DD_HTTPX_SERVICE
     """
-    code = """
-import asyncio
-import sys
+    import asyncio
+    import sys
 
-import httpx
+    import httpx
 
-from ddtrace.contrib.httpx import patch
-from tests.contrib.httpx.test_httpx import get_url
-from tests.utils import snapshot_context
+    from ddtrace.contrib.httpx import patch
+    from tests.contrib.httpx.test_httpx import get_url
+    from tests.utils import snapshot_context
 
-patch()
-url = get_url("/status/200")
+    patch()
+    url = get_url("/status/200")
 
-async def test():
-    token = "tests.contrib.httpx.test_httpx.test_configure_service_name_env"
-    with snapshot_context(token=token):
-        httpx.get(url)
+    async def test():
+        token = "tests.contrib.httpx.test_httpx.test_configure_service_name_env"
+        with snapshot_context(token=token):
+            httpx.get(url)
 
-    with snapshot_context(token=token):
-        async with httpx.AsyncClient() as client:
-            await client.get(url)
+        with snapshot_context(token=token):
+            async with httpx.AsyncClient() as client:
+                await client.get(url)
 
-if sys.version_info >= (3, 7, 0):
-    asyncio.run(test())
-else:
-    asyncio.get_event_loop().run_until_complete(test())
-    """
-    env = os.environ.copy()
-    env["DD_HTTPX_SERVICE"] = "env-overridden-service-name"
-    env["DD_SERVICE"] = "global-service-name"
-    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
-    assert status == 0, err
-    assert err == b""
+    if sys.version_info >= (3, 7, 0):
```

Comment:
Would be cool to follow this PR up with some more magic to allow inline running of async code so we don't need this boilerplate every time.